src/lib/fragments/AgGrid.react.js

```diff
@@ -1042,6 +1049,38 @@ export default class DashAgGrid extends Component {
         }
     }
 
+    scrollTo(reset = false) {
+        const {gridApi} = this.state;
+        const {scrollTo, setProps} = this.props;
+        if (!gridApi) {
+            return;
+        }
+        const rowPosition = scrollTo.rowPosition ? scrollTo.rowPosition : 'top';
+        if (scrollTo.rowIndex || scrollTo.rowIndex === 0) {
+            gridApi.ensureIndexVisible(scrollTo.rowIndex, rowPosition);
+        } else if (scrollTo.rowId) {
+            const node = gridApi.getRowNode(scrollTo.rowId);
+            gridApi.ensureNodeVisible(node, rowPosition);
```

Comment:
I'd probably have `rowId` take precedence over `rowIndex`, though I don't know why you'd provide both. Maybe you want to get a certain row but it might not exist you want an index to fall back on?

also `rowId` could be falsy, so `if (scrollTo.rowId !== undefined)` perhaps?

Reply:
My thought process on the priority of `rowIndex`, `rowId`, and `data` was to go from more to less specific (`rowIndex` always have unique values, but `rowId` can be set to have non-unique values). I think that the use case for using both deliberately is pretty niche, but with the current implementation it wouldn't fall back on any index/rowId because of the `if-else`.

Reply:
What happens if the grid can't scroll to a row? I think it returns false. 

Could use the scrolled idea from your other suggestion @alexcjohnson.

So, instead of doing if-else. They are all if with the additional condition of if it scrolls. 

rowIds being non-unique is an issue that causes a great deal of pain, even outside of scrolling. Sorting creates multiple lines, etc. We should expect the rowId to be unique.

Reply:
Yeah OK - I was thinking about things like you've filtered out the row you're trying to scroll to, maybe instead you want to scroll to the top - so that would be ID (or data) falling back on index. But reflecting on this, most likely in that case I'd rather the grid not scroll at all, in which case this doesn't matter.